app/src/modules/location/location.c

```diff
@@ -107,6 +120,27 @@ void location_task(void)
 			return;
 		}
 
+		if (&NETWORK_CHAN == chan) {
+			LOG_DBG("Network status received");
+			enum network_status status ;
+			err = zbus_chan_read(&NETWORK_CHAN, &status, K_SECONDS(1));
+			if (err) {
+				LOG_ERR("Failed to read network status: %d", err);
+				SEND_FATAL_ERROR();
+				return;
+			}
+
+			if (status == NETWORK_CONNECTED) {
+				/* GNSS has to be enabled after the modem is initialized and enabled */
+				err = lte_lc_func_mode_set(LTE_LC_FUNC_MODE_ACTIVATE_GNSS);
```

Comment:
This function only needs to be called once. Having a guard here saves few cycles in the long run. No big deal though.